alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/processor/service/CycleService.java

```diff
@@ -0,0 +1,127 @@
+package eu.xenit.alfresco.processor.service;
+
+import eu.xenit.alfresco.processor.model.Cycle;
+import eu.xenit.alfresco.processor.util.DateTimeUtil;
+import lombok.AllArgsConstructor;
+import org.alfresco.repo.solr.Transaction;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@AllArgsConstructor
+public class CycleService {
+    private static final Logger logger = LoggerFactory.getLogger(CycleService.class);
+
+    protected final RetryingTransactionHelper retryingTransactionHelper;
+
+    public void execute(HealthProcessorConfiguration configurationService) {
+        Cycle cycle = createCycle( configurationService);
+        AtomicBoolean continueCycle = new AtomicBoolean();
+        run(configurationService, cycle, continueCycle);
+        while(configurationService.isEnabled()
+                && !configurationService.isRunOnce()
+                && continueCycle.get()) {
+            run(configurationService, cycle, continueCycle);
+        }
+    }
+
+    Cycle createCycle(HealthProcessorConfiguration configurationService) {
+        return new Cycle(
+                configurationService.getTransactionLimit(),
```

Comment:
these parameters don't coincide anymore with the variables since the last commit